downloadservice/app.py

```diff
@@ -159,10 +183,94 @@ def login(user):
     return render_template('index.html', user=user, token=token)
 
 
-def get_upstream_session():
+def certificate_validity(certificate):
+    try:
+        x509 = OpenSSL.crypto.load_certificate(
+            OpenSSL.crypto.FILETYPE_PEM, certificate)
```

Comment:
does this raise something on expired on incorrectly signed certificate?

Reply:
Yes, it does and I transform the exception into a readable one for the user.

Reply:
then it probably raises something more specific than `Exception`, it's good to catch that